assert/assertions.go

```diff
@@ -1864,17 +1864,20 @@ func Eventually(t TestingT, condition func() bool, waitFor time.Duration, tick t
 
 // CollectT implements the TestingT interface and collects all errors.
 type CollectT struct {
+	failed bool
 	errors []error
 }
 
 // Errorf collects the error.
 func (c *CollectT) Errorf(format string, args ...interface{}) {
+	c.failed = true
 	c.errors = append(c.errors, fmt.Errorf(format, args...))
 }
 
-// FailNow panics.
-func (*CollectT) FailNow() {
-	panic("Assertion failed")
+// FailNow stops the execution by calling runtime.Goexit.
+func (c *CollectT) FailNow() {
+	c.failed = true
```

Comment:
Add `if c.errors == nil { c.errors = []errors{} }`. That will allow to make the `failed` boolean equivalent to `c.errors != nil` and allow to avoid adding `failed`. That will make the patch much smaller.

Reply:
I don't mind there being an explicit flag for `failed` -- it makes it a bit clearer rather than using `nil` as an implicit sentinel value meaning "failed."

Reply:
We will make it clear by adding `fail()` and `failed() bool` methods.